### DISCUSSION TITLE
Mirror - not found packages for sync.

### DISCUSSION THREAD

**Discussion by Bashev** [vtsykun/packeton#114] [General]
I'm trying to configure this cool new feature, but i'm stacked, because i not understand how to proceed.

```[repo@hippo ~]$ php bin/console packagist:sync:mirrors magento -vvv
                                                                                                                        
 [INFO] Start sync: https://repo.magento.com                                                                                                                                                                                                 

Loading root file packages.json
Checked CA file /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem: valid
Using HTTP basic authentication with username "0ef73f0384286943833241edd2356651"
Downloading https://repo.magento.com/packages.json
[200] https://repo.magento.com/packages.json
Loading provider includes...
    0 [--->------------------------] < 1 sec 4.0 MiBSync enabled packages via composer v1 api https://repo.magento.com/p/%package%$%hash%.json
Not found enabled packages for re sync
```

or 
```
Not found any provider-includes in packages.json
```

**Comment by vtsykun** [comment]
Hi, this is expected behavior and debug messages.
If the metadata has not changed and there are not manually enabled packages -  sync was skipped

**Reply by Bashev** [reply]
I'm agree, but i have no idea how to enable manually some packages.

If i understand correctly if i make compose require magento/magento-community-package and on my composer.json is only mirror repository, this will fetch it and install it, but never happens. 

Probably I understood it wrong.

**Comment by vtsykun** [comment]
Hi, By default packages are automatically mirrored and enabled when it requested by composer `update/require command`. You may see enabled/uses packages in the grid

![Selection_1461](https://user-images.githubusercontent.com/21358010/235709127-a75785be-ecff-498b-af5f-5a61622d2302.png)

**Reply by Bashev** [reply]
This is the reason to open the topic.

my composer.json
```
{
    "require": {
        "magento/module-customer": "*"
    },"repositories": [{
        "type": "composer",
        "url": "https://repo.company.com/mirror/magento"
    }]
}
```
when run composer update, i receive an error - package cannot be found.

I trying to debug it, and saw, i have no keys in the redis where to packages need to be added.

![image](https://user-images.githubusercontent.com/1854269/235712189-0983a928-e220-4134-b94a-4834ef7b8cf9.png)

--- 
Some additional findings (all related to redis data if i understand correctly).

![image](https://user-images.githubusercontent.com/1854269/235719823-aa0a5b38-b504-4572-9579-e3b7545a2aef.png)

![image](https://user-images.githubusercontent.com/1854269/235720115-dbec1d2e-014e-46bf-b0aa-3081d67c65a9.png)

**Reply by vtsykun** [reply]
Looks as bug, you can try to disable a lazy sync to avoid not found error and preload all packages into cache.

As I see `repo.magento.com` support only v1 `https://repo.magento.com/p/%package%$%hash%.json` format where hash may change when requested packages and trigger 404 error

```yaml
        magento:
            url: https://repo.magento.com
            lazy: false
```

Also can you provider example of root metadata format `https://repo.magento.com/packages.json` for debug? Just is it private, but I do not have magento account

**Reply by Bashev** [reply]
I sent credentials to: tsykun314 at gmail.

**Reply by vtsykun** [reply]
Fixed #115.
